### PR TITLE
feat: filter url parameters

### LINKED ISSUE #893: bug: Filters are not kept when user refreshes the page
## Bug Definition
Filters are not kept when user refreshes the page

-- Please add discord contact information
** discord username: @notDepresseDeveloper#0919 **

## Bug environment
Describe the environment produces the bug. For example:
rc.afetharita.com
afetharita.com

## Describe how you are producing the bug step by step

1. Go to afetharita.com or rc.afetharita.com
2. Select something from filters
3. Refresh the page
4. Bug appears

## Expected Behaviour
Filters have to be cached and user should see previous filter they choose.

## Desktop Information
- Operating System: Windows 10, 11
- Browser: Chrome, Safari

## Mobile Phone Information
- Operating System (with the version): Android, IOS 
- Browser : Chrome, Safari (for IOS), Samsung Browser (Android)

### PR DESCRIPTION
**Pull request by snturk**
# Description

**discord username: Tundra#9805**

**closes #893 **

Filter can now be able to get relevant URL parameters like 'lastTime=last30Minutes' and when a filter change, a parameter added to url to share the link.

## Things to check before creating a PR
- [X] I have inspected my topic, checked.
- [X] If it is a core feature, executed detailed tests.

## Creating PR rules
- [ ] PR must be created for an issue with approved tag. Otherwise PR will be rejected.
- [ ] Relevant issue number: The issue number related to PR must be attached to head of PR header, after prefix after prefix # must be attached in parenthesis. A header like this could be used "prefix(#issue_number): PR header" .
- [ ] A descriptive and understandable title: The PR title should clearly describe the nature and purpose of the changes. The PR title should be the first thing displayed when the PR is opened. And it should follow the semantic commit rules. For example, a title like "docs(#issueId): Add README.md" can be used.
- [ ] Related file selection: Only relevant files should be touched and no other files should be affected.
- [ ] Format and Lint Suitability : The code should be made in accordance with a certain format standard and examined according to the lint rules.
- [ ] Clean commit history: The commits where the changes are made should be clear and organized.
- [ ] Opening PR when job is completed: PR should be opened when job is completed and sent for review by other team members.

## Changes

- [X] Filter related components (ChannelFilterMenu, CommonFilterMenu, FilterTimeMenu, ReasonFilterMenu) improved with url parameters
- [X] addParameterToUrl and getParameterValueFromUrl util functions added

# How were these changes tested?

I've tried various combinations of URL parameters, no problem.

Example Link: `LOCALHOST/en?lat=38.094574707566274&lng=35.211181640625&zoom=7&lastTime=last3Hours&reasons=erzak%2Cyemek&channel=twitter`

**Test Configuration**:

* Firmware version:
* Hardware:
* Toolchain:
* SDK:

### CODE REVIEW

File: src/components/UI/FilterMenu/CommonFilterMenu.tsx [hunk 1]
```diff
@@ -37,9 +40,18 @@ function CommonFilterMenu<ValueType>({
   const handleMenuItemClick = (event: MouseEvent<HTMLLIElement>) => {
     const value = (event.currentTarget.dataset.value ?? null) as ValueType;
     setSelectedValue(value);
+    addParameterToUrl(
+      parameterName,
+      (event.currentTarget.dataset.value ?? null) as string
```
Comment (eraygundogmus) [inline src/components/UI/FilterMenu/CommonFilterMenu.tsx:45]: i think it is not meaningful to set a null as string

File: src/components/UI/FilterMenu/ReasonFilterMenu.tsx [hunk 2]
```diff
@@ -29,6 +30,7 @@ export const ReasonFilterMenu: React.FC = () => {
   const [filterValues, setValues] = React.useState<string[]>(
     reasonFilterMenuOptions
   );
+  const parameterName = "reasons";
```
Comment (eraygundogmus) [inline src/components/UI/FilterMenu/ReasonFilterMenu.tsx:33]: same here as above

File: src/utils/helpers.ts [hunk 3]
```diff
@@ -5,4 +5,15 @@ const capitalize = (str: string) => {
   return str.charAt(0).toUpperCase() + str.slice(1);
 };
 
-export { capitalize, isNaN };
+function addParameterToUrl(param: string, value: string) {
```
Comment (eraygundogmus) [inline src/utils/helpers.ts:8]: `addParameterToURL
`
`getParameterValueFromURL`

**Review by eraygundogmus — CHANGES_REQUESTED** [review]

**Comment by eraygundogmus** [comment]
Meanwhile, I can see your deployment has failed. To fix it

- yarn build 
- find errors
- fix
- commit and push

File: src/components/UI/FilterMenu/CommonFilterMenu.tsx (hunk 1 above)
Comment (snturk) [reply src/components/UI/FilterMenu/CommonFilterMenu.tsx]: fixed on commit a3cb348c2e28945e4ace5269648f8523807b427b

File: src/utils/helpers.ts (hunk 3 above)
Comment (snturk) [reply src/utils/helpers.ts]: fixed on commit a3cb348c2e28945e4ace5269648f8523807b427b

**Comment by snturk** [comment]
> Meanwhile, I can see your deployment has failed. To fix it
> 
> * yarn build
> * find errors
> * fix
> * commit and push


Done, no errors occurred.

**Review by efegurkan — DISMISSED** [review]
Seems good to me after the changes. Someone else also please take a look.

**Review by efegurkan — COMMENTED** [review]
Seems good, in general. Thank you for your contribution <3 

There is one case that I figured after the deployment. 
- If we select a few reason filters they are added to URL as expected.
- At first refresh filters are set on page load as expected.
- Once page load is done, even though filters are set. They are not on the url anymore.

I tried with following
`https://deprem-yardim-frontend-dd8asetrr-afet.vercel.app/en?lat=37.535557424752696&lng=36.89483642578126&zoom=8&reasons=erzak%2Chayvanlar-icin-tedavi%2Ckonaklama%2Ckurtarma%2Clojistik%2Csu%2Cyagma%2Cyemek`

If you refresh page twice, filters are gone from url.
`https://deprem-yardim-frontend-dd8asetrr-afet.vercel.app/en?lat=37.535557424752696&lng=36.89483642578126&zoom=8`

I revoked my approval, sorry for any inconveniences I made. 

Deployment seems to have issues with translations but I don't think it is related to this PR :thinking:

**Comment by snturk** [comment]
Some other params like `lat` or `zoom` was overriding the existing ones. So commit 17b4bc72b1ca046bb766849611c4444f26f32f5e fixes that issue with keeping the existing ones.

File: src/components/UI/FilterMenu/ReasonFilterMenu.tsx (hunk 2 above)
Comment (snturk) [reply src/components/UI/FilterMenu/ReasonFilterMenu.tsx:33]: Didn't understand :( Is there a problem related to line 33?

**Comment by usirin** [comment]
we are doing a big refactor & redesign ty so much for the contribution and sorry for not taking this in.

**Pull request closed without merging**